vllm/engine/arg_utils.py

```diff
@@ -1,12 +1,112 @@
 import argparse
 import dataclasses
+import torch
+import io
+import os
+import typing
+
 from dataclasses import dataclass
-from typing import Optional, Tuple
+from typing import Optional, Tuple, Union, Callable, Any
 
 from vllm.config import (CacheConfig, ModelConfig, ParallelConfig,
                          SchedulerConfig)
 
 
+@dataclass
+class TensorizerArgs:
+    download_dir: Union[
+        io.BufferedIOBase,
+        io.RawIOBase,
+        typing.BinaryIO,
+        str,
+        bytes,
+        os.PathLike,
+        int,
+    ]
+    device: Optional[Union[torch.device, str]] = None
+    dtype: Optional[torch.dtype] = None
+    serialize: bool = False
+    serializer_encryption: Optional[bool] = False
+    lazy_load: bool = False
+    plaid_mode: bool = False
+    plaid_mode_buffers: Optional[int] = None
+    verify_hash: bool = False
+    filter_func: Optional[Callable[[str], Union[bool, Any]]] = None
+    deserializer_encryption_key: Optional[str] = None
+
+    def __post_init__(self):
+        self.file_obj = self.download_dir
+        self.serializer_params = (
+            self.file_obj,
+            None,  ## Placeholder for `compress_tensors`
+            self.serializer_encryption
+        )
+
+        self.deserializer_params = (
+            self.file_obj,
+            self.device,
+            self.filter_func,
+            self.dtype,
+            self.lazy_load,
+            self.plaid_mode,
+            self.plaid_mode_buffers,
+            self.verify_hash,
+            self.deserializer_encryption_key
+        )
+
+    @staticmethod
+    def add_cli_args(
+            parser: argparse.ArgumentParser) -> argparse.ArgumentParser:
+        """Tensorizer CLI arguments"""
+        parser.add_argument(
+            "--serialize",
+            action='store_true',
+            help="In the event that serialized weights can't be found at"
```

Comment:
Seems like all help descriptions are the same.

Reply:
Yep, I'm going to get to that